torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -542,7 +546,32 @@ void ProcessGroupNCCL::ncclCommWatchdogInternal() {
     watchdogCV_.wait_for(
         lock,
         std::chrono::milliseconds(kWatchdogThreadSleepMillis),
-        [&]() -> bool { return terminateWatchdog_.load(); });
+        [&]() -> bool { return terminateProcessGroup_.load(); });
+  }
+}
+
+void ProcessGroupNCCL::workCleanupLoop() {
+  while (!terminateProcessGroup_.load()) {
+    std::unique_lock<std::mutex> lock(workVectorMutex_);
+    // We busy-poll the work vector every kWatchdogThreadSleepMillis
+    // milliseconds as long as the atomic is True.
+    workVectorCV_.wait_for(
+        lock,
+        std::chrono::milliseconds(kWatchdogThreadSleepMillis),
+        [&]() -> bool { return terminateProcessGroup_.load(); });
```

Comment:
Is this correct? This means this won't wake up until `terminateProcessGroup_` is true, even if there are notifications?

Reply:
Yeah I think we didn't notify the `workVectorCV` from enqueue in an earlier version - seems like it's best to remove it entirely and always sleep for the specified duration instead